test/firebase_auth_mocks_test.dart

```diff
@@ -570,6 +577,51 @@ void main() {
     );
   });
 
+  // Temp variable to hold tUser before provider is linked
+  final _unlinkedTUser = tUser;
+
+  test('User.linkWithProvider works if not already linked', () async {
+    final auth = MockFirebaseAuth(signedIn: true, mockUser: tUser);
+    final user = auth.currentUser;
+    final provider = TwitterAuthProvider();
+    final cred = await user!.linkWithProvider(provider);
+    expect(cred, isNotNull);
+    expect(
+      user.providerData.any((info) => info.providerId == 'twitter.com'),
+      true,
+    );
+  });
+
+  test('User.linkWithProvider throws Exception if already linked', () async {
+    // tUser is still linked to twitter.com from previous test
+    final auth = MockFirebaseAuth(signedIn: true, mockUser: tUser);
+    final user = auth.currentUser;
+    final provider = TwitterAuthProvider();
+    expect(
+      () async => await user!.linkWithProvider(provider),
+      throwsA(isA<FirebaseAuthException>()),
+    );
+  });
+
+  test('User.unlink succeeds', () async {
+    final auth = MockFirebaseAuth(signedIn: true, mockUser: tUser);
+    final user = auth.currentUser;
+    final provider = TwitterAuthProvider();
+    final newUser = await user!.unlink(provider.providerId);
+    expect(newUser, _unlinkedTUser);
```

Comment:
This won't work for two reasons:
- _unlinkedTUser refers to tUser and user!.unlink returns `this`, which also refers to tUser. No matter the content, this expectation will always be true.
- even if _unlinkedTUser and tUser refered to different users, MockUser.props does not use providerData for the equals comparison: https://github.com/atn832/firebase_auth_mocks/blob/86a03529791c7d3dd2a90920810063666ee098a2/lib/src/mock_user.dart#L147-L156.

The cleanest way to check that unlink works is to compare user's providerData to [].

Reply:
I fixed this test in #94 and fixed MockUser.equals in 582aba75ae469fb82c257415f071097dca4ae509.